File: elixir-resend-examples/examples/inbound.exs
Mix.install([
  {:resend, "~> 0.4.5"},
  {:dotenvy, "~> 0.8"}
])

if File.exists?(".env"), do: Dotenvy.source!([".env"])

api_key = System.get_env("RESEND_API_KEY") || raise "RESEND_API_KEY environment variable is required"

# This example fetches a previously received inbound email by ID.
# The email ID comes from an "email.received" webhook event payload.
# Set up inbound webhooks at: https://resend.com/webhooks
email_id = System.get_env("INBOUND_EMAIL_ID") || "example-email-id"

if email_id == "example-email-id" do
  IO.puts("Note: Set INBOUND_EMAIL_ID to fetch a real inbound email.")
  IO.puts("You get this ID from the 'email.received' webhook event.\n")
end

case Resend.Emails.get(email_id) do
  {:ok, email} ->
    IO.puts("=== Inbound Email Details ===")
    IO.puts("From: #{email["from"]}")
    IO.puts("To: #{Enum.join(email["to"] || [], ", ")}")
    IO.puts("Subject: #{email["subject"]}")
    IO.puts("Created: #{email["created_at"]}")

    case email["text"] do
      text when is_binary(text) and byte_size(text) > 0 ->
        preview =
          if String.length(text) > 200,
            do: String.slice(text, 0, 200) <> "...",
            else: text

        IO.puts("\nText preview:\n#{preview}")

      _ ->
        :ok
    end

    IO.puts("\nDone!")

  {:error, reason} ->
    IO.puts(:stderr, "Error fetching email: #{inspect(reason)}")
    System.halt(1)
end

File: dotnet-resend-examples/Examples/Inbound.cs
namespace ResendExamples;

using Resend;

public static class Inbound
{
    public static async Task RunAsync()
    {
        var apiKey = Environment.GetEnvironmentVariable("RESEND_API_KEY")
            ?? throw new Exception("RESEND_API_KEY environment variable is required");

        var client = ResendClient.Create(apiKey);

        // This example fetches a previously received inbound email by ID.
        // The email ID comes from an "email.received" webhook event payload.
        // Set up inbound webhooks at: https://resend.com/webhooks
        var emailId = Environment.GetEnvironmentVariable("INBOUND_EMAIL_ID") ?? "example-email-id";

        if (emailId == "example-email-id")
        {
            Console.WriteLine("Note: Set INBOUND_EMAIL_ID to fetch a real inbound email.");
            Console.WriteLine("You get this ID from the 'email.received' webhook event.\n");
        }

        var email = (await client.ReceivedEmailRetrieveAsync(Guid.Parse(emailId))).Content;

        Console.WriteLine("=== Inbound Email Details ===");
        Console.WriteLine($"From: {email.From}");
        Console.WriteLine($"To: {string.Join(", ", email.To)}");
        Console.WriteLine($"Subject: {email.Subject}");
        Console.WriteLine($"Created: {email.MomentCreated}");

        if (!string.IsNullOrEmpty(email.TextBody))
        {
            var preview = email.TextBody.Length > 200 ? email.TextBody[..200] + "..." : email.TextBody;
            Console.WriteLine($"\nText preview:\n{preview}");
        }

        Console.WriteLine("\nDone!");
    }
}
